### DISCUSSION TITLE
Home Page Design Idea

### DISCUSSION THREAD

**Discussion by Perseusdehond** [jellyfin/jellyfin-vue#498] [UI & UX]
I really like Jellyfin and especially the new Vue client. I would like to give again an concept I had to enchance the Vue client in the future. This design introduces some new features for faster access and for people who can't decide what to watch just like me ;).

![Home Page Design](https://user-images.githubusercontent.com/42120256/103565520-c1f9e800-4ec0-11eb-8a90-ec5be5092838.jpg)

This is quite a stretched design, so if some parts of this design get used, you will probably only see 2 parts of the recommendations. I will once again explain what each part contains so my idea becomes clear. Even though it is for the most part pretty self explainatory.

1. Because of the multiple folders of content that could take up a lot of space with their recent/latest content. Especially with the sections below. It would be a good idea and less waiste of space to bundle multiple (especially movies and series) together in recent content. There probably has to be an option which folders can be displayed in this section.
2. This section shows recommended movies based on your latest watched movies. This could show your latest watched movie a bit bigger at the left or take the traditional aproach with a: because you watched 'recent movie name' so the poster doesn't have to be shown. (I like it a bit more with the bigger poster). Behind the posters you see different collors that the transparent back I tried to go with, so I think you should look at section two with a black background for this instance. This you can also see in sections 4 and 5, as the movies there are displayed with a black background, but they don't let the poster or gray background through. It is just there to give a rough idea :)
3. These two sections are to show off the traditional way of displaying movies, but just categorized based on genre. This one is probably the easiest to implement, because it uses the 'latest movies' display, just with a different filter.
4. This section shows how a collection could look at the home screen. So a big landscaped poster at the back with it fading away to the right is quite nice. I got this idea from another design that also used a bigger poster at the back to display movies a bit in this way.
5. This section shows movies based on an actor you've watched a movie from if that actor has multiple movies to show. I got this design from HBO max. It shows the actor in a circle, but to get this to look nice, I had to put another background color behind it. I used the same color as the side bar, because this is easy for the eyes.
6. This section show randomly generated suggestions based on whatever filters there are. These are just some shortcuts to have that take up only a little of the screenspace, but I think it would add a nice touch.
7. This section could only really work if you were able to make collections yourself for content you like. Then you can add set collection to the homescreen and it could look something like this. This idea is loosely based on how Disney+ filters their content, but with more customization in mind. I think it is something unique and I quite like the idea of making your own collections and also being able of displaying some permanently on the home screen. I got the designs from [CoverLabs](https://www.coverlabs.io). That site is normally used to make nice posters with the same design.
8.  This would be a feature. How it works is like this: you go to your movies/series. Then you choose all you filters you want to apply and the you could heart/star the filter with an heart/star-icon on the right. (I don't know if this is clear, because I didn't make something visual for this). After you hearted/star-ed this filter, it would pop up onder the folder you filtered it in. To still make the left bar concise, you could add an arrow to fold in and out these sections. From this menu you could also unheart them to remove them. This is seen in section 8.
9. Because there is more of a focus on recommending content to users in this design, you could also add the option for a randomized button. This gives you a few options (3 or something) to choose between with trailers to watch if available. I haven't made a visual design of this, just a rough idea. I also understand that this may result in too much buttons at the top, but that depends on what the devs find important to put there.
10. This isn't really a section, but I would like a different design for the arrows. Something like the old jellyfin web arrows would look a bit nicer in my opinion. These current arrows are just a bit big. But I wouldn't mind either way.

Wheww, that was quite the list. I was thinking of such a design and features because the people that use my Jellyfin server are more used to the main stream streaming services. This keeps them from using Jellyfin more often and I think that is a shame. That's why I tried to design how some features could look in Jellyfin and they seem to quite like the design so far.

For this design I didn't use landscape posters, because the current Jellyfin design also doesn't use them and I didn't want to complete throw away that design. Although it looks a bit worse, especially for the collections design (section 4), because the background posters doesn't really get shown of more. These posters obscure it a bit more, but I still like the design enough in this way. Some designs are also not finished, as I was not sure on how many movies you should recommend at such a page. That is why you see recommendations shift between 6, 7 and 8 movies. It is more to show off what could work or maybe after seeing this couldn't work. I also didn't really design this for series episodes and music in mind. Although with slight tweeks these sections could work just as well for those content types. I am not saying that these features have to be implemented, but I would love to see some of them coming over. Just take a look and see if you like it ;)

**Comment by heyhippari** [comment]
This is really nice and has some good ideas :)

We'd need to see if there are APIs to generate all of these in a nice way, but I like it!

**Reply by Perseusdehond** [reply]
Thank you, I understand that it isn't that easy to implement, but I hope there are some APIs to generate some of my section ideas in a nice way in the future

**Comment by ferferga** [comment]
Hello @Perseusdehond!

First of all, thank you very much for your ideas. We (I say we as that's how we usually handled past discussions about concepts in the past but, in reality, I'm talking about me) didn't reply to your previous Collections discussion because we (I) liked it and there was nothing useful to add. We even discussed about it the other day in our Client Dev rooms in Matrix which, if you haven't already, I suggest you to join, as most of the development talks happens there as it's quicker than here. So, if for a moment, you thought that it was completely ignored, don't :).

As @MrTimscampi points out, this is going to be more difficult than your collections idea, as it will probably require backend changes. But we have them here so we can get some inspiration for improvements at some point, as there's still plenty to do with the home page (see future plans of #455 for instance).

**Reply by ferferga** [reply]
@Perseusdehond Are you a graphic designer, UI/UX expert btw?

**Reply by Perseusdehond** [reply]
@ferferga Thank you for your reply. I am happy you like some of my ideas and I already joined the Matrix channel, thnx for the invite ;). I just tought that this discussion feature would be better to look back at some design ideas if needed. I am already pretty actively following this project and I will join some discussions in Matrix when I think I can add something (same username and image as on Github).

I also understand the collections idea would be easier to implement, because I mostly used the already existing Vue elements to make that design, to keep it simple and in-line with what is currently there and also because it wouldn't be too much work to make if you wanted to use my design as an example.

And I am not a graphic desinger or UI/UX expert btw. I mostly do this as a side hobby, because I like to design stuff and I am busy with my study for a big chunk of the week. But designing all kinds of stuff is something I enjoy to do when I have some spare time. So could very well be that I just create another idea when I have the time ;). If there is something on the design aspect you are not entirely sure about, you may always ask me to take a look at what I should do. You don't have to ;). I'm already glad if I could help with some inspiration to make this project even better :)

**Comment by JonathanTreffler** [comment]
I can't wait to see such a highly customizeable UI 😃
